### PR TITLE
backend: Remove experimental Langchain chat/multihop

### PR DESCRIPTION
**Pull request by tianjing-li**
**AI Description**

This PR introduces several changes to the Cohere toolkit, including updates to documentation, authentication, deployment, and feature configuration.

## Documentation
- The `CONTRIBUTING.md` file has been updated to provide clearer instructions for contributors, including a new section on how to add features.
- The `README.md` file has been modified to reflect the toolkit's capabilities and deployment options.
- A new file, `docs/custom_tool_guides/tool_auth_guide.md`, has been added to provide guidance on adding authentication to custom tools.
- The `docs/how_to_guides.md` file has been updated to include a section on how to add your own chat interface.
- The `docs/theming.md` file has been updated to provide instructions on changing the logo, favicon, and metadata.

## Authentication
- The `docs/auth_guide.md` file has been updated to provide instructions on setting up authentication strategies in the `configuration.yaml` file.
- The `src/backend/config/settings.py` file has been modified to remove the `use_experimental_langchain` field from the `FeatureFlags` class.
- The `src/backend/routers/auth.py` file has been updated to remove a note about tool authentication being experimental.

## Deployment
- The `docs/deployment_guides/aws_ecs_single_container.md` file has been updated to provide more detailed instructions for deploying the toolkit to AWS ECS.
- The `docs/deployment_guides/aws_ecs_single_container_ec2.md` file has been updated to provide more detailed instructions for deploying the toolkit to AWS ECS EC2.
- The `docs/deployment_guides/gcp_deployment.md` file has been updated to provide more detailed instructions for deploying the toolkit to Google Cloud Platform.
- The `docs/deployment_guides/gcp_one_click_deployment.md` file has been updated to provide more detailed instructions for deploying the toolkit to Google Cloud Platform using a one-click deployment.
- The `docs/deployment_guides/single_container.md` file has been updated to provide more detailed instructions for building and starting the toolkit using a single container setup.
- The `docs/github_migrations_action.md` file has been updated to provide more detailed instructions for setting up a GitHub Action to automatically run database migrations.

## Feature Configuration
- The `.env-template` file has been updated to remove the `USE_EXPERIMENTAL_LANGCHAIN` field.
- The `docs/config_details/config_description.md` file has been updated to remove the `use_experimental_langchain` field from the `feature_flags` section.
- The `src/backend/config/configuration.template.yaml` file has been updated to remove the `use_experimental_langchain` field from the `feature_flags` section.
- The `src/backend/config/settings.py` file has been updated to remove the `use_experimental_langchain` field from the `FeatureFlags` class.
- The `src/backend/config/tools.py` file has been updated to remove the `langchain_tools` list and the `use_langchain_tools` variable.
- The `src/backend/routers/chat.py` file has been updated to remove the `LangChainChat` class and the `langchain_chat_stream` function.
- The `src/backend/routers/experimental_features.py` file has been updated to remove the `USE_EXPERIMENTAL_LANGCHAIN` field from the `experimental_features` dictionary.
- The `src/backend/chat/custom/langchain.py` file has been removed.
- The `src/backend/chat/custom/custom.py` file has been updated to remove a reference to the `StreamEvent` class.

## Summary
This PR introduces changes to the Cohere toolkit's documentation, authentication, deployment, and feature configuration. The updates aim to improve the clarity and usability of the toolkit, while also removing experimental features related to LangChain.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,32 +1,41 @@
 # OSS Toolkit Contributing Guide
 
-Thank you for your interest in contributing to Cohere's OSS toolkit. This guide will help explain the contribution workflow from opening an issue, creating a PR, to reviewing and merging your added changes. For development setup, please refer to the [README](README.md).
+Thank you for your interest in contributing to Cohere's OSS toolkit. This guide covers the contribution workflow, from opening an issue to reviewing and merging changes. For development setup, please refer to the [README](README.md).
```
Comment (lusmoura) [inline CONTRIBUTING.md:3]: Are these changes the same as #792?

**Review by lusmoura — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (tianjing-li) [reply CONTRIBUTING.md:3]: Yep thats correct, just merged the other one in

File: src/interfaces/coral_web/src/components/Settings/SettingsTab.tsx [hunk 2]
```diff
@@ -36,16 +33,6 @@ export const SettingsTab: React.FC = () => {
     });
   };
 
-  if (isLangchainModeOn) {
-    return (
-      <div className="flex items-center justify-center px-5">
-        <Text styleAs="p-lg" className="select-none text-center text-volcanic-100">
-          {STRINGS.langchainMultihopSettingsDisabledDescription}
```
Comment (tomtobac) [inline src/interfaces/coral_web/src/components/Settings/SettingsTab.tsx:43]: probably we can remove this variable from the `STRINGS` constant

**Review by tomtobac — APPROVED** [review]
LGTM 🔥

File: src/interfaces/coral_web/src/components/Settings/SettingsTab.tsx (hunk 2 above)
Comment (tianjing-li) [reply src/interfaces/coral_web/src/components/Settings/SettingsTab.tsx:43]: Sounds good

**Pull request merged**